integrations/influxdb_connector/influxdb-timestream-connector/src/records_builder.rs

```diff
@@ -12,168 +12,202 @@ pub enum SchemaType {
     SingleTableMultiMeasure,
 }
 
-#[tracing::instrument(skip_all, level = tracing::Level::TRACE)]
-pub fn get_builder(schema: SchemaType, measure_name: String) -> impl BuildRecords {
-    match schema {
-        SchemaType::SingleTableMultiMeasure => {
-            return multi_measure_builder::MultiMeasureBuilder {
-                measure_name: None,
-                schema_type: SchemaType::SingleTableMultiMeasure,
-            }
-        }
-        SchemaType::MultiTableMultiMeasure => {
-            return multi_measure_builder::MultiMeasureBuilder {
-                measure_name: Some(measure_name.to_string()),
-                schema_type: SchemaType::MultiTableMultiMeasure,
-            }
-        }
-    }
+/// A group of records sharing common attributes.
+#[derive(Debug, Clone)]
+pub struct AttributeGroupedRecords {
+    /// A Record comprised of a measure name, dimensions, and timestamp precision.
+    /// All records within the "records" field share these attributes.
+    pub common_attributes: timestream_types::Record,
+    /// Records containing measure values and timestamps. These records share
+    /// attributes contained in the "common_attributes" field.
+    pub records: Vec<timestream_types::Record>,
 }
 
-#[tracing::instrument(skip_all, level = tracing::Level::TRACE)]
-pub fn build_records(
-    records_builder: &impl BuildRecords,
-    metrics: &[Metric],
-    precision: &timestream_types::TimeUnit,
-) -> Result<HashMap<String, Vec<timestream_types::Record>>, Error> {
-    records_builder.build_records(metrics, precision)
-}
-
-#[tracing::instrument(skip_all, level = tracing::Level::TRACE)]
-pub fn table_creation_enabled() -> Result<bool, Error> {
-    // Convert the env var table_creation_enabled to bool
-
-    match std::env::var("enable_table_creation") {
-        Ok(enabled) => Ok(env_var_to_bool(enabled)),
-        Err(_) => Err(anyhow!(
-            "enable_table_creation environment variable is not defined"
-        )),
+impl AttributeGroupedRecords {
+    pub fn len(&self) -> usize {
+        self.records.len()
     }
-}
 
-#[tracing::instrument(skip_all, level = tracing::Level::TRACE)]
-pub fn database_creation_enabled() -> Result<bool, Error> {
-    // Convert the env var database_creation_enabled to bool
-    match std::env::var("enable_database_creation") {
-        Ok(enabled) => Ok(env_var_to_bool(enabled)),
-        Err(_) => Err(anyhow!(
-            "enable_database_creation environment variable is not defined"
-        )),
+    pub fn is_empty(&self) -> bool {
+        self.records.is_empty()
     }
 }
 
-#[tracing::instrument(skip_all, level = tracing::Level::TRACE)]
-pub fn env_var_to_bool(env_var: String) -> bool {
-    // Convert the env var to bool
-
-    matches!(env_var.as_str(), "true" | "t" | "1")
+/// A single Record split into its common attributes and measure value(s).
+#[derive(Debug, Clone)]
+pub struct RecordPair {
```

Comment:
Is there any benefit for using a struct rather than a tuple?

Reply:
I'm fine with whichever you prefer. I think something like `common_records.0/1` is also not too unclear.